catalyst/contrib/dl/callbacks/kornia_transform.py

```diff
@@ -0,0 +1,142 @@
+from typing import Dict, Optional, Sequence, Tuple, Union
+
+from kornia.augmentation import AugmentationBase
+import torch
+from torch import nn
+
+from catalyst.contrib.registry import TRANSFORMS
+from catalyst.core.callback import Callback, CallbackNode, CallbackOrder
+from catalyst.core.runner import IRunner
+
+
+class BatchTransformCallback(Callback):
+    """Callback to perform data augmentations on GPU using kornia library.
+
+    Look at `Kornia: an Open Source Differentiable Computer Vision
+    Library for PyTorch`_ for details.
+
+    To apply augmentations only during specific loader e.g. only during
+    training :class:`catalyst.core.callbacks.control_flow.ControlFlowCallback`
+    callback can be used. For config API it can look like this:
```

Comment:
what do you think about some minimal example for Notebook API too?
we could adapt our MNIST example easily, I think
```python

import os
import torch
from torch.nn import functional as F
from torch.utils.data import DataLoader
from catalyst import dl
from catalyst.contrib.data.transforms import ToTensor
from catalyst.contrib.datasets import MNIST
from catalyst.utils import metrics

model = torch.nn.Linear(28 * 28, 10)
optimizer = torch.optim.Adam(model.parameters(), lr=0.02)

loaders = {
    "train": DataLoader(MNIST(os.getcwd(), train=True, download=True, transform=ToTensor()), batch_size=32),
    "valid": DataLoader(MNIST(os.getcwd(), train=False, download=True, transform=ToTensor()), batch_size=32),
}

class CustomRunner(dl.Runner):

    def predict_batch(self, batch):
        # model inference step
        return self.model(batch[0].to(self.device).view(batch[0].size(0), -1))

    def _handle_batch(self, batch):
        # model train/valid step
        x, y = batch
        y_hat = self.model(x.view(x.size(0), -1))

        loss = F.cross_entropy(y_hat, y)
        accuracy01, accuracy03 = metrics.accuracy(y_hat, y, topk=(1, 3))
        self.batch_metrics.update(
            {"loss": loss, "accuracy01": accuracy01, "accuracy03": accuracy03}
        )

        if self.is_train_loader:
            loss.backward()
            self.optimizer.step()
            self.optimizer.zero_grad()

runner = CustomRunner()
# model training
runner.train(
    model=model,
    optimizer=optimizer,
    loaders=loaders,
    logdir="./logs",
    num_epochs=5,
    verbose=True,
)
```

Reply:
done